### DISCUSSION TITLE
How to find touchscreen resolution in Win10 tablet?

### DISCUSSION THREAD

**Discussion by paluigi** [onitake/gsl-firmware#160] [Q&A]
Hello,
I need some help to understand how to set the touchscreen resolution parameter when generating the Linux touchscreen driver for a tablet.

Longer story:
I have a low end tablet ( UnisCom MZ89-IPS ...it's old and made in China for the domestic market, I was able to get no info online), and I want to install Linux.
I extracted the touchscreen drivers following the process indicated in the Readme (gslx680_ts_acpi style, as I'm testing with a Lubuntu live image at the moment), and with the new driver the touchscreen works...but needs a lot of calibration (and rotation), and it's not that accurate.
To generate the [gslx680_ts_acpi](https://github.com/onitake/gslx680-acpi) drivers i used the following parameters, because I saw the tablet screen resolution is 800x1200 (in portrait mode):

`tools/fwtool -c firmware.fw -m 1680 -w 800 -h 1280 -t 10 silead_ts.fw`

**I see many other tablets listed here have used touchscreen size parameters different from screen resolution...is it only a matter of trial&error, or there is a way where I can check the actual parameters for my touchscreen size?**

Once this is fixed, I can complete the testing and start a pull request for the new firmware. Thanks!

**Comment by onitake** [comment]
> I see many other tablets listed here have used touchscreen size parameters different from screen resolution...is it only a matter of trial&error, or there is a way where I can check the actual parameters for my touchscreen size?

Sadly, if we were able to do this, we wouldn't need that complicated procedure. :disappointed: 
I'm sure there is some way to extract that information from the device-specific firmware image, but it's not documented anywhere and nobody has any clue where to find it.
